design/replace-CLI-with-SDK.md

```diff
@@ -0,0 +1,220 @@
+
+<!-- toc -->
+- [Motivation](#motivation)
+- [Scope](#scope)
+- [High Level Design](#high-level-design)
+  - [Kopia SDK wrappers](#kopia-sdk-wrappers)
+    - [Storage pkg](#storage-pkg)
+    - [Repository pkg](#repository-pkg)
+  - [Repository server controller changes](#repository-server-controller-changes)
+	- [Kopia CLI Approach](#kopia-cli-approach)
+	- [Kopia SDK Approach](#kopia-sdk-approach)
+<!-- /toc -->
+
+
+# Motivation
+
+Kanister uses a kubernetes custom controller makes use of kopia as a primary backup and restore tool. 
+The detailed design of the customer controller can be found [here](https://github.com/kanisterio/kanister/blob/master/design/kanister-kopia-integration.md)
+
+The custom controller called as repository server controller currently uses kopia CLI to perform the kopia operations. All the
+operations are executed inside a pod using the `kubectl exec` function. `kubectl exec` can be flaky for long running operations
+and gives less control over command that is being executed. 
+
+The goal over here is to build a kopia server programatically and reduce the dependency on kopia CLI in turn reducing the usage of 
+`kubectl exec` by using kopia SDK and gain more flexibility over the operations
+
+## Scope
+1. Implement a library that provides wraps the SDK functions provided by kopia to connect to underlying storage providers - S3, Azure, GCP, Filestore
+2. Build another pkg on top of the library implemented in #1 that can be used to perform repository operations
+3. Modify the repository server controller to run a pod that executes a custom image. The binary would take all the
+necessary steps to make the kopia server ready using kopia SDK and kopia CLI. 
+
+## High Level Design
+
+### Kopia SDK Wrappers
+
+#### Storage pkg
```

Comment:
We don't have to include implementation details in the proposal. Just defining interfaces should suffice.